### DISCUSSION TITLE
When does the revalidation process begin?

### DISCUSSION THREAD

**Discussion by xuesichao** [react-hook-form/react-hook-form#9504] [Q&A] [answered]
Hi, I'm adding validation to a multi-page component (Wizard) which is not based on HTML `<form>`. I want the revalidation for current page to start happening after I click the `Next Step` button for each step of Wizard. Right now I have one big yup schema for entire Wizard form fields. However the revalidation never happens `onChange` event when there's invalid input and after I click `Next Step` button.

I checked the documentation about `reValidateMode` where it says:

>This option allows you to configure validation strategy when inputs with errors get re-validated **after a user submits the form (onSubmit event). By default, validation is triggered during the input change event.**

For my application, the form `onSubmit` event only get triggered in the last step and I think that's why the react hook form revalidation is not take effect.

I plan to split my yup schema into steps and use `useForm` for each step for my Wizard, in this case I can get a `handleSubmit` for each step. 

I have a clarification that when will revalidation process begin?
1. After HTML `<form>` tag's `onSubmit` event get triggered
2. After `handleSubmit` method of `useForm()` get triggered

Your documentation indicates option 1, but I feel like option 2 makes more sense. Could you please help me?

**Comment by bluebill1049** [comment, marked as answer]
can you share a codesandbox? have you tried changing `mode`?

**Reply by xuesichao** [reply]
Yes, I tried every mode, and for each of them the validation/revalidation on change is not working.

Could you tell me the answer of this question?
>I have a clarification that when will revalidation process begin?
>
>1. After HTML `<form>` tag's `onSubmit` event get triggered
>2. After `handleSubmit` method of `useForm()` get triggered

**Reply by bluebill1049** [reply]
`revalidation` happen after the first submit

**Reply by xuesichao** [reply]
Thanks! So If I'm not using a `<form>` in my component, it is expected that revalidation can never be triggered?

**Reply by bluebill1049** [reply]
if `onSubmit` never happened, then it wouldn't kick in

**Reply by bluebill1049** [reply]
more precise `handleSubmit` function

**Reply by xuesichao** [reply]
Great, that answers my question. I feel like there's a need to clarify this in the documentation:
<img width="915" alt="image" src="https://user-images.githubusercontent.com/20471438/206303177-9573b6ce-a35a-4d50-b214-9aac8db896ba.png">
https://react-hook-form.com/api/useform

Right now it says `form (onSubmit event)` but actually it is `handleSubmit` method of `useForm()`.

Anyway really appreciate your help, we can close this.
